### PR TITLE
Update profiling docs.

### PR DESCRIPTION
**Pull request by ltratt**
In particular, the Firefox profiler seems (at least on light usage) to be a good fit with profiling, so update the docs while I remember how to use this stuff.

Note that this does assume the changes in https://github.com/ykjit/yk/pull/1016 have merged.

### CODE REVIEW

File: docs/src/dev/profiling.md [hunk 1]
```diff
@@ -60,47 +60,80 @@ Fields and their meaning are as follows:
 
 ## Perf
 
-Note that `yk-config --cflags` includes `-Wl,--no-rosegment`, which [gives
-better profiling
-information](https://github.com/flamegraph-rs/flamegraph#cargo-flamegraph) for
-binaries linked with `lld` (as all yk C interpreters are). 
+On Linux, `perf` can be used to profile yk. You must first need to record an
```
Comment (vext01) [inline docs/src/dev/profiling.md:63]: s/must//

Comment (ltratt) [reply docs/src/dev/profiling.md]: Fixed in dc170ce.

File: docs/src/dev/profiling.md [hunk 2]
```diff
@@ -60,47 +60,80 @@ Fields and their meaning are as follows:
...
+```
+$ cargo build --profile=release-with-debug
+```
+
+Ensure that the interpreter you are profiling links to the appropriate version
+of yk and then call `perf record`:
 
 ```
 $ perf record --call-graph dwarf -g ./interpreter ...args...
-$ perf report -G --no-inline
```
Comment (vext01) [inline docs/src/dev/profiling.md:72]: Did you mean to uninvert the call graph?

Did you mean to use bare symbols (and not DWARF symbols, which in my experience gives much better profiles)?

Comment (ltratt) [reply docs/src/dev/profiling.md:72]: Je ne parlez pas votres question.

Comment (vext01) [reply docs/src/dev/profiling.md:72]: Second part of my comment was nonsense.

Comment (ltratt) [reply docs/src/dev/profiling.md:72]: Ah, `-G` was in the original doc and I copied it blindly. Fixed in f690220.

File: docs/src/dev/profiling.md [hunk 3]
```diff
@@ -60,47 +60,80 @@ Fields and their meaning are as follows:
...
 
-`--call-graph=dwarf` makes perf use DWARF debugging information, which gives
-better quality profiling data than the default, but requires that you compile
-with debugging info.
+This uses `--call-graph dwarf` to force perf use DWARF debugging information:
+this will only be useful if you have compiled yk with embedded debugging
+information, as recommended above.
+
 
-We recommend passing `--no-inline` since, for all but the smallest runs,
```
Comment (vext01) [inline docs/src/dev/profiling.md:79]: It's fast now?

Comment (ltratt) [reply docs/src/dev/profiling.md:79]: Not on b16... Note that this part has just moved elsewhere, so the warning is still present.

Comment (vext01) [reply docs/src/dev/profiling.md:79]: I see it moved.

**Comment by vext01** [comment]
Firefox profiler looks neat.

**Comment by vext01** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**